File: greedy/SingleStockTrader.cpp
class SingleStockTrader {
public:
    int maxProfit(vector<int>& prices) {
        int maxPrft = 0;
        int minPrice = INT_MAX;
        for (int i = 0; i < prices.size(); i++) {
            maxPrft = max(maxPrft, prices[i] - minPrice);
            minPrice = min(minPrice, prices[i]);
        }
        return maxPrft;
    }
};

File: greedy/SingleStockTrader.java

public class SingleStockTrader {
    public int maxProfit(int[] prices) {
        if (prices == null) {
            return 0;
        }
        int maxPrft = 0;
        int lowPrice = Integer.MAX_VALUE;
        for (int i = 0; i < prices.length; i++) {
            lowPrice = Math.min(lowPrice, prices[i]);
            maxPrft = Math.max(maxPrft, prices[i] - lowPrice);
        }
        return maxPrft;
    }
}
